extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -429,11 +436,49 @@ export class BookTocManager implements IBookTocManager {
 		}
 	}
 
+	public async addNewNotebook(notebookName: string, notebookPath: string, fileType: FileType, book?: BookModel): Promise<void> {
+		if (fileType === FileType.Notebook) {
+			await this.createNotebook(notebookPath.concat(fileType), undefined);
+		} else {
+			await fs.writeFile(notebookPath.concat(fileType), '');
+		}
+		if (book) {
+			let toc = yaml.safeLoad((await fs.readFile(book.tableOfContentsPath, 'utf8')));
+			let notebookToc: JupyterBookSection = {
+				title: notebookName,
+				file: path.posix.join(path.posix.sep, notebookName)
+			};
+			if (book.version === BookVersion.v1) {
+				notebookToc = convertTo(book.version, notebookToc);
+			}
+			toc.push(notebookToc);
+			await fs.writeFile(book.tableOfContentsPath, yaml.safeDump(toc, { lineWidth: Infinity, noRefs: true, skipInvalid: true }));
```

Comment:
It would be a lot better I think if the operation to read/write the TOC was all handled through the same logic - because then you could more easily control what is read and written. Right now this is just adding to the possibility of race conditions if something else is also modifying the TOC at the same time.